### ISSUE TITLE
No early stopping in the deepinv.Trainer

### ISSUE THREAD

**Issue by Melvin-klein** [deepinv/deepinv#436]
It would be great to support early stopping in the `deepinv.Trainer`.

**Comment by tachella** [comment]
hey Melvin, many thanks for this suggestion!

I'm not sure I fully understand your point, how would the trainer be early-stopped?

**Comment by tachella** [comment]
what do you mean? we already have a max number of training steps

**Comment by Andrewwango** [comment]
Maybe do you mean to stop training when (`train` or `test`) `loss` or `metric` reaches `tol` instead of when `epoch` reaches `epochs`?

**Comment by Melvin-klein** [comment]
Thanks for your answers ! I'm new in the ML field so I hope what I'm saying makes sense : I was thinking about stopping the training based on performance on a validation set. For example, if the performance measured on the validation set hasn't decreased for `n` epochs, we stop the training and keep the model learned `n` epochs before stopping.

**Comment by tachella** [comment]
Thanks for the clarification, we are currently working on this!

**Comment by Andrewwango** [comment]
Closed by #437
